### ISSUE TITLE
Add MaNo Scorer to metrics.py as a realistic unsupervised scorer

### ISSUE THREAD

**Issue by ambroiseodt** [scikit-adaptation/skada#288]
Following @rflamary's advice, it could be interesting to see whether [MaNo](https://arxiv.org/pdf/2405.18979) can be used as a realistic unsupervised scorer.

The method was designed for Unsupervised Accuracy Estimation and shows a linear correlation with the test OOD accuracy without needing test labels or training data/labels. Hence, one could in principle use it as a realistic unsupervised scorer.

*Disclaimer*
- I am an author of [MaNo](https://arxiv.org/pdf/2405.18979) (NeurIPS 2024)
- It was tailored for computer vision classification tasks with large-scale deep models, hence it is not guaranteed at all that it can be useful for other modalities
- In particular, it relies on test logits obtained after an inference step of the neural network. The implementation in Skada should/could work both for shallow and deep methods and thus the original implementation should be adapted
- As the unsupervised setting is particularly hard, two hyperparameters are needed in MaNo (although fixed and quite stable in the original experiments). This might also be an issue for its efficiency/versatility to other modalities and with smaller neural networks
